### PR TITLE
Add support for refresh-inhibited parameter

### PR DESCRIPTION
**Pull request by sergio-costas**
This PR implements in snapd-glib support for the 'refresh-inhibited' select parameter in the GET /v2/snap method, from the SD167 document ( https://docs.google.com/document/d/1HJQWKzgaB3yPKwRUqlxYhgyaKG5IcJe8eZOT4YY1CI8 )

### CODE REVIEW

File: snapd-glib/snapd-snap.c [hunk 1]
```diff
@@ -424,6 +426,13 @@ snapd_snap_get_hold (SnapdSnap *self)
 }
 
 
+GDateTime *
+snapd_snap_get_proceed_time (SnapdSnap *self)
+{
+    g_return_val_if_fail (SNAPD_IS_SNAP (self), 0);
```
Comment (robert-ancell) [inline snapd-glib/snapd-snap.c:432]: 0 -> NULL

File: snapd-glib/snapd-snap.c [hunk 2]
```diff
@@ -424,6 +426,13 @@ snapd_snap_get_hold (SnapdSnap *self)
 }
 
 
+GDateTime *
```
Comment (robert-ancell) [inline snapd-glib/snapd-snap.c:429]: Missing documentation

File: snapd-qt/Snapd/snap.h [hunk 3]
```diff
@@ -86,6 +87,7 @@ class Q_DECL_EXPORT QSnapdSnap : public QSnapdWrappedObject
     bool devmode () const;
     qint64 downloadSize () const;
     QDateTime hold () const;
+    QDateTime proceed_time () const;
```
Comment (robert-ancell) [inline snapd-qt/Snapd/snap.h:90]: proceed_time -> proceedTime

File: snapd-glib/snapd-snap.c (hunk 1 above)
Comment (sergio-costas) [reply snapd-glib/snapd-snap.c:432]: Good catch! Fixed.

File: snapd-glib/snapd-snap.c (hunk 2 above)
Comment (sergio-costas) [reply snapd-glib/snapd-snap.c:429]: Ops... you are right.

BTW: I see in all the documentation that it uses "@snap: a @SnapdSnap", but the parameter name in the function is `SnapdSnap *self`... Shouldn't it be "@self: a @SnapdSnap"?

File: snapd-qt/Snapd/snap.h (hunk 3 above)
Comment (sergio-costas) [reply snapd-qt/Snapd/snap.h]: fixed!

File: snapd-glib/snapd-snap.c (hunk 2 above)
Comment (sergio-costas) [reply snapd-glib/snapd-snap.c:440]: Fixed.

Comment (robert-ancell) [reply snapd-glib/snapd-snap.c:440]: Responded in the other PR - the C files use 'self' and the headers/docs use the specific name.

File: snapd-glib/snapd-client.c [hunk 4]
```diff
@@ -1951,13 +1951,17 @@ void
...
+                              GCancellable *cancellable,
+                              GAsyncReadyCallback callback,
+                              gpointer user_data)
 {
     g_return_if_fail (SNAPD_IS_CLIENT (self));
 
     g_autoptr(SnapdGetSnaps) request = _snapd_get_snaps_new (cancellable, names, callback, user_data);
     if ((flags & SNAPD_GET_SNAPS_FLAGS_INCLUDE_INACTIVE) != 0)
         _snapd_get_snaps_set_select (request, "all");
+    if ((flags & SNAPD_GET_SNAPS_FLAGS_REFRESH_INHIBITED) != 0)
```
Comment (robert-ancell) [inline snapd-glib/snapd-client.c:1963]: These two flags are now mutually exclusive - should assert they're both not set.

File: snapd-glib/snapd-client-sync.c [hunk 5]
```diff
@@ -512,6 +512,10 @@ G_GNUC_END_IGNORE_DEPRECATIONS
  * If @flags contains %SNAPD_GET_SNAPS_FLAGS_INCLUDE_INACTIVE then also return snaps
  * with status %SNAPD_SNAP_STATUS_INSTALLED.
  *
+ * If @flags contains %SNAPD_GET_SNAPS_FLAGS_REFRESH_INHIBITED, then it will return
```
Comment (robert-ancell) [inline snapd-glib/snapd-client-sync.c:515]: Document that only one of SNAPD_GET_SNAPS_FLAGS_INCLUDE_INACTIVE or SNAPD_GET_SNAPS_FLAGS_REFRESH_INHIBITED can be set.

File: snapd-glib/snapd-snap.c [hunk 6]
```diff
@@ -423,6 +425,24 @@ snapd_snap_get_hold (SnapdSnap *self)
     return self->hold;
 }
 
+/**
+ * snapd_snap_get_proceed_time:
+ * @self: a @SnapdSnap
```
Comment (robert-ancell) [inline snapd-glib/snapd-snap.c:430]: @self->@snap

File: snapd-glib/snapd-client.c (hunk 4 above)
Comment (sergio-costas) [reply snapd-glib/snapd-client.c:1969]: Added a check.

File: snapd-glib/snapd-client-sync.c (hunk 5 above)
Comment (sergio-costas) [reply snapd-glib/snapd-client-sync.c:515]: Done.

Comment (sergio-costas) [reply snapd-glib/snapd-client-sync.c:515]: Mmm... but should I add a GError parameter, and set it if both flags are set?

File: snapd-glib/snapd-snap.c (hunk 6 above)
Comment (sergio-costas) [reply snapd-glib/snapd-snap.c:430]: Yes, now I understand it.

Comment (sergio-costas) [reply snapd-glib/snapd-snap.c]: Done.

File: snapd-glib/snapd-client-sync.c (hunk 5 above)
Comment (robert-ancell) [reply snapd-glib/snapd-client-sync.c:515]: I think an assert would be more appropriate - treat it as a programmer error to set both flags.

Comment (sergio-costas) [reply snapd-glib/snapd-client-sync.c:515]: You are right. Done.

**Comment by robert-ancell** [comment]
@sergio-costas is this ready to land?

**Review by robert-ancell — APPROVED** [review]

**Pull request merged**